packages/app/src/debug/GroupedDebugFailedTest.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div
+    v-for="(group, j) in props.groups"
+    :key="j"
+    class="flex flex-col flex-start w-full pl-16px justify-center grouped-row-class hover:bg-gray-50"
+    data-cy="grouped-row"
+  >
+    <div
+      class="flex flex-start flex-row font-normal border-b-gray-100 h-12 text-base text-gray-700 gap-x-2.5 items-center non-italic"
+      data-cy="test-failed-metadata"
+    >
+      <StatsMetadata
+        :order="['GROUP_NAME', 'OS', 'BROWSER']"
+        :groups="[group]"
+        :group-name="group.groupName!"
+      />
+      <div
+        data-cy="debug-artifacts"
+        class="flex flex-grow space-x-4.5 opacity-0 pr-18px justify-end grouped-row-artifacts"
+      >
+        <div
+          v-for="artifact, l in debugArtifacts[group.id]"
+          :key="l"
+          :data-cy="`artifact-${artifact.icon}`"
+        >
+          <DebugArtifactLink
+            :icon="artifact.icon"
+            :popper-text="artifact.text"
+            :url="artifact.url"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts" setup>
+import type { TestResults } from './DebugSpec.vue'
+import StatsMetadata from './StatsMetadata.vue'
+import DebugArtifactLink from './DebugArtifactLink.vue'
+import { computed } from 'vue'
+import type { StatsMetadata_GroupsFragment } from '../generated/graphql'
+
+const props = defineProps<{
+  failedTests: TestResults[]
+  groups: StatsMetadata_GroupsFragment[]
+}>()
+
+const debugArtifacts = computed(() => {
+  return props.failedTests.reduce<{[groupID: string]: {icon: string, text: string, url: string | null | undefined }[] }>((acc, curr) => {
+    //TODO Update logic to not rely on defaulting to empty strings
```

Comment:
Is this TODO a style preference or is does it functionally make a difference to something? If you don't plan to make this change, we can remove the TODO and maybe leave a comment explaining. If it's an important change, we should make an issue and put the issue number in the TODO.

Reply:
@amehta265 I just realized that this is rendering all three buttons every time, but it should only render if the corresponding boolean value in the result is true and the url is present.